### PR TITLE
Screen orientation: don't assume -primary/secondary are supported

### PR DESCRIPTION
**Pull request by marcoscaceres**
Fixed a potential timeout in "lock-sandboxed-iframe.html". 

Not all implementations support every type of lock.

### CODE REVIEW

File: screen-orientation/onchange-event.html [hunk 1]
```diff
@@ -9,11 +9,12 @@
     try {
       await document.exitFullscreen();
     } catch (e) {}
+    screen.orientation.unlock();
   });
   await test_driver.bless("request full screen", () => {
     return document.documentElement.requestFullscreen();
   });
-  const type = screen.orientation.type;
+  const type = screen.orientation.type.startsWith("portrait") ? "landscape" : "portrait";
```
Comment (cdumez) [inline screen-orientation/onchange-event.html:17]: Isn't this reversed?

File: screen-orientation/resources/sandboxed-iframe-locking.html [hunk 2]
```diff
@@ -5,17 +5,17 @@
...
 
   let msg = "";
   try {
-    await screen.orientation.lock("portrait-primary")
+    await document.documentElement.requestFullscreen();
+    await screen.orientation.lock("portrait")
     msg = screen.orientation.type;
   } catch (error) {
-    msg = error.name;
+    msg = `error: ${error.name} ${error.message}`;
```
Comment (marcoscaceres) [inline screen-orientation/resources/sandboxed-iframe-locking.html:18]:
```suggestion
    msg = error.toString();
```

File: screen-orientation/resources/iframe-listen-orientation-change.html [hunk 3]
```diff
@@ -1,5 +1,9 @@
 <script>
-window.screen.orientation.addEventListener('change', () => {
-  parent.window.postMessage(screen.orientation.type, "*");
-});
+  try {
+    window.screen.orientation.addEventListener("change", () => {
+      parent.window.postMessage(screen.orientation.type, "*");
+    });
+  } catch (err) {
+    parent.window.postMessage(err.message, "*");
```
Comment (marcoscaceres) [inline screen-orientation/resources/iframe-listen-orientation-change.html:7]:
```suggestion
    parent.window.postMessage(err.toString(), "*");
```

File: screen-orientation/resources/iframe-listen-orientation-change.html [hunk 4]
```diff
@@ -1,5 +1,9 @@
 <script>
-window.screen.orientation.addEventListener('change', () => {
-  parent.window.postMessage(screen.orientation.type, "*");
-});
+  try {
```
Comment (cdumez) [inline screen-orientation/resources/iframe-listen-orientation-change.html:2]: What can throw here?

File: screen-orientation/onchange-event.html (hunk 1 above)
Comment (marcoscaceres) [inline screen-orientation/onchange-event.html:17]: good catch! 

```suggestion
  const type = screen.orientation.type.startsWith("portrait") ? "portrait" : "landscape";
```

File: screen-orientation/resources/iframe-listen-orientation-change.html (hunk 4 above)
Comment (marcoscaceres) [reply screen-orientation/resources/iframe-listen-orientation-change.html:2]: `screen.orientation.type` being `undefined`

Comment (marcoscaceres) [inline screen-orientation/resources/iframe-listen-orientation-change.html:2]:
```suggestion
  // try/catch orientation being undefined...  
  try {
```

**Review by cdumez — APPROVED** [review]
LGTM

**Pull request merged**